src/units/types.cpp

```diff
@@ -1394,6 +1394,26 @@ void unit_type::apply_scenario_fix(const config& cfg)
 			boost::remove_erase(advances_to_, str);
 		}
 	}
+	// support advancement AMLA tags here
+	// TODO: I think I badly screwed up the if condition here
+	// disclaimer: I dont how to code cpp, I know Python and SQL.
+	if (auto attr = cfg.child_range("advancement")) {
```

Comment:
You want has_child, not child_range.

Also, I don't see any code here that actually adds the advancement. You're adding to the "abilities that can be obtained by advancements" list, but the advancement itself hasn't been added.

Reply:
> You're adding to the "abilities that can be obtained by advancements" list, but the advancement itself hasn't been added.

What does an adding advancement line look like? (I don't know Cpp)

Reply:
Look at this https://github.com/wesnoth/wesnoth/pull/9353